### PR TITLE
experimental: wasm/pyodide support

### PR DESCRIPTION
**Pull request by mscolnick**

### CODE REVIEW

File: marimo/_runtime/runtime.py [hunk 1]
```diff
@@ -1079,6 +1079,36 @@ def instantiate(self, request: CreationRequest) -> None:
...
+            FunctionCallResult(
+                function_call_id=request.function_call_id,
+                return_value=ret,
+                status=status,
+            ).broadcast()
+            CompletedRun().broadcast()
+        elif isinstance(request, DeleteRequest):
+            self.delete(request)
+        elif isinstance(request, StopRequest):
+            return None
```
Comment (akshayka) [inline marimo/_runtime/runtime.py:1108]: `StopRequest` should make the Kernel's control loop quit. This is used for stopping kernels in run mode.

File: marimo/_runtime/runtime.py [hunk 2]
```diff
@@ -1214,34 +1244,10 @@ def sigterm_handler(signum: int, frame: Any) -> None:
...
-                status=status,
-            ).broadcast()
-            CompletedRun().broadcast()
-        elif isinstance(request, DeleteRequest):
-            kernel.delete(request)
-        elif isinstance(request, StopRequest):
-            break
-        else:
-            raise ValueError(f"Unknown request {request}")
+        kernel.handle_message(request)
```
Comment (akshayka) [inline marimo/_runtime/runtime.py:1247]: Should break out of this loop on a `StopRequest`

File: marimo/_utils/typed_connection.py [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+if TYPE_CHECKING:
+    from multiprocessing.connection import Connection
 
 T = TypeVar("T")
 
 
-class TypedConnection(Generic[T], Connection):
+class TypedConnection(Generic[T]):
```
Comment (akshayka) [inline marimo/_utils/typed_connection.py:12]: Why the change to this class? I didn't see it used in the `pyodide` code path

**Review by akshayka — APPROVED** [review]
LGTM except for Playwright

File: marimo/_utils/typed_connection.py (hunk 3 above)
Comment (mscolnick) [reply marimo/_utils/typed_connection.py:12]: i wasnt able to mock out `Connection` and this gets used at file/module import, instead of lazily. this refactor just made it more lazy (and switching to composition over inheritance)
